root/static/scripts/relationship-editor/common/dialog.js

```diff
@@ -195,6 +196,27 @@ const RE = MB.relationshipEditor = MB.relationshipEditor || {};
 
             this.changeOtherRelationshipCredits = {source: ko.observable(false), target: ko.observable(false)};
             this.selectedRelationshipCredits = {source: ko.observable('all'), target: ko.observable('all')};
+
+            this.tooShortBeginYearError = debounce(function () {
+                const relationship = self.relationship();
+                if (!relationship.begin_date) {
+                    return '';
+                }
+                const date = relationship.begin_date;
+                const valid = dates.isYearFourDigits(date.year());
+                return valid ? '' : l('The year should have four digits. If you want to enter a year earlier than 1000 CE, please pad with zeros, such as “0123”.');
+            });
+
+            this.tooShortEndYearError = debounce(function () {
```

Comment:
It'd be nice to merge these definitions though. One way is we can have a `tooShortYear(date)` function right above these and call it inside both debounces, passing `self.relationship().begin_date` or `self.relationship().end_date`.